app/uk/gov/hmrc/merchandiseinbaggage/service/CustomsDutyCalculator.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ */
+
+package uk.gov.hmrc.merchandiseinbaggage.service
+
+import java.time.LocalDate
+
+import cats.data.EitherT
+import cats.instances.future._
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import uk.gov.hmrc.merchandiseinbaggage.connectors.CurrencyConversionConnector
+import uk.gov.hmrc.merchandiseinbaggage.model.api.{CalculationRequest, CurrencyConversionResponse}
+import uk.gov.hmrc.merchandiseinbaggage.model.core.AmountInPence._
+import uk.gov.hmrc.merchandiseinbaggage.model.core.{AmountInPence, BusinessError, CurrencyNotFound}
+import uk.gov.hmrc.merchandiseinbaggage.repositories.CustomsRate._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+trait CustomsDutyCalculator extends CurrencyConversionConnector {
+
+  def customDuty(httpClient: HttpClient, calculationRequest: CalculationRequest)
+                (implicit hc: HeaderCarrier, ec: ExecutionContext): EitherT[Future, BusinessError, AmountInPence] =
+    for {
+      lt         <- EitherT.liftF(findCurrencyRate(httpClient, calculationRequest.currency, LocalDate.now))
```

Comment:
Do we know enough about the currency api to rely on this (will there always be a collection of rates for LocalDate.now)?

Reply:
I have tried many different things and always had a response
However if API can't find rate will return a 299
If you enter an invalid currency code you will still get a 200 but without rate....so I am handle that on our side